[libpq] lack of pdb files in package for static library linkage

When building libpq with static library linkage pdb files aren't copied into output lib directory that results in warning on linking. (if you don't have that pdb avaiable in build path on the same computer).

Changing in
ports\libpq\patches\windows\minimize_install.patch
+ "$conf\\", "libpgcommon\\libpgcommon.lib",
to
+		"$conf\\", "libpgcommon\\libpgcommon.lib", "libpgcommon\\libpgcommon.pdb", "libpq\\libpq.pdb", "libpgport\\libpgport.pdb",

solves the problem (but I doubt that it's the correct way).

**Environment**
- OS: Windows 10
- Compiler: VS 2019 

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install libpq:x86-windows-static
2. ./vcpkg vcpkg export libpq:x86-windows-static --nuget --output=vcpkg --nuget-version=1.1.0.0
Import resulting package to VS project and build the resulting project.

If you don't have pdb file in package (or in directory where package was built on the same computer where you are building your project) you get warnings during build like
6>libpq.lib(encnames.obj) : warning LNK4099: PDB 'libpq.pdb' was not found with 'libpq.lib(encnames.obj)' or at '...\libpq.pdb'; linking object as if no debug info

Try renaming your vcpkg directory and building the project as described in problem.

--- a/ports/libpq/CONTROL
+++ b/ports/libpq/CONTROL
@@ -1,6 +1,6 @@
 Source: libpq
 Version: 12.2
-Port-Version: 9
+Port-Version: 10
 Build-Depends: libpq[core,bonjour] (osx)
 Supports: !uwp
 Homepage: https://www.postgresql.org/

--- a/ports/libpq/patches/windows/fix-compile-flag-Zi.patch
+++ b/ports/libpq/patches/windows/fix-compile-flag-Zi.patch
@@ -11,3 +11,23 @@ index b93992f..2397511 100644
        <CompileAs>Default</CompileAs>
      </ClCompile>
      <Link>
+diff --git a/src/tools/msvc/Install.pm b/src/tools/msvc/Install.pm
+index e0c9a88..98f9e67 100644
+--- a/src/tools/msvc/Install.pm
++++ b/src/tools/msvc/Install.pm
+@@ -341,9 +341,12 @@ sub CopySolutionOutput
+ 				  || croak "Could not copy $pf.$ext\n";
+ 			}
+ 		}
+-		lcopy("$conf\\$pf\\$pf.pdb", "$target\\bin\\$pf.pdb")
+-		  || croak "Could not copy $pf.pdb\n";
+-		print ".";
++        if ($1 eq 'DynamicLibrary')
++        {
++            lcopy("$conf\\$pf\\$pf.pdb", "$target\\bin\\$pf.pdb")
++            || croak "Could not copy $pf.pdb\n";
++            print ".";
++        }
+ 	}
+ 	print "\n";
+ 	return;

--- a/ports/libpq/patches/windows/fix-compile-flag-Zi.patch
+++ b/ports/libpq/patches/windows/fix-compile-flag-Zi.patch
@@ -1,0 +1,13 @@
+diff --git a/src/tools/msvc/MSBuildProject.pm b/src/tools/msvc/MSBuildProject.pm
+index b93992f..2397511 100644
+--- a/src/tools/msvc/MSBuildProject.pm
++++ b/src/tools/msvc/MSBuildProject.pm
+@@ -333,7 +333,7 @@ sub WriteItemDefinitionGroup
+       <BrowseInformation>false</BrowseInformation>
+       <WarningLevel>Level3</WarningLevel>
+       <SuppressStartupBanner>true</SuppressStartupBanner>
+-      <DebugInformationFormat>ProgramDatabase</DebugInformationFormat>
++      <DebugInformationFormat>OldStyle</DebugInformationFormat>
+       <CompileAs>Default</CompileAs>
+     </ClCompile>
+     <Link>

--- a/ports/libpq/portfile.cmake
+++ b/ports/libpq/portfile.cmake
@@ -42,6 +42,7 @@ set(PATCHES
         patches/windows/MSBuildProject_fix_gendef_perl.patch
         patches/windows/msgfmt.patch
         patches/windows/python_lib.patch
+        patches/windows/fix-compile-flag-Zi.patch
         patches/linux/configure.patch)
 
 if(VCPKG_LIBRARY_LINKAGE STREQUAL static)

